velox/exec/ExchangeSource.h

```diff
@@ -76,18 +76,29 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   /// Requests the producer to generate up to 'maxBytes' more data and reply
   /// within 'maxWait'. Returns a future that completes when producer responds
   /// either with 'data' or with a message indicating that all data has been
-  /// already produced or data will take more time to produce.
+  /// already produced or data will take more time to produce. Automatically
+  /// acknowledges successful reception of the data buffers received earlier.
   virtual folly::SemiFuture<Response> request(
       uint32_t maxBytes,
       std::chrono::microseconds maxWait) = 0;
 
   /// Ask for available data sizes that can be fetched.  Normally should not
   /// fetching any actual data (i.e. Response::bytes should be 0).  However for
   /// backward compatibility (e.g. communicating with coordinator), we allow
-  /// small data (1MB) to be returned.
+  /// small data (1MB) to be returned. Automatically acknowledges successful
+  /// reception of the data buffers received earlier.
   virtual folly::SemiFuture<Response> requestDataSizes(
       std::chrono::microseconds maxWait) = 0;
 
+  /// Acknowledge successful reception of the data buffers received earlier
+  /// without requesting any more data. The engine will try to minimize the
+  /// number of acknowledge requests. For example if there is enough buffer
+  /// memory available to send an another data request right away an explicit
+  /// acknowledge will not be issued.
+  virtual ContinueFuture acknowledge() {
```

Comment:
It feels a bit strange to not have an ID or token or something like that in an 'ack', but the indent of the API makes sense to me. The semantics here might be described as 'pause', e.g. the engine says that it needs some time to process already received data and may not request more for a while.

Reply:
Discussed offline. Indeed "pause" better describes what this method does. Changed the name to pause and changed the comment. Thank you for the suggestion.

Could you please take an another look?